src/components/UserList.js

```diff
@@ -1,23 +1,67 @@
 import React from 'react';
 import UserItem from './UserItem';
+import { addUser } from "../redux/actions";
+import { connect } from "react-redux";
 
 class UserList extends React.Component {
-    render() {
-        return (
-            <>
-                <form>
-                    <div>
-                        <input />
-                        <input type="submit" value="dodaj" />
-                    </div>
-                </form>
-
-                <ul>
-                    <UserItem />
-                </ul>
-            </>
-        )
-    }
+	state = {
+		inputValue: "",
+	};
+
+	handleInputChange = e => {
+		this.setState({ inputValue: e.target.value });
+	};
+
+	handleSubmit = e => {
+		e.preventDefault();
+
+		const { users } = this.props;
+		const { inputValue } = this.state;
+
+		const newUser = {
+			name: inputValue,
+			id: users.length + 1,
```

Comment:
Bezpieczniej jest używać `uuid()` - zobacz: że jeśli masz 2 użytkowników o id=1 oraz id=2. 
Potem usuwasz tego o id=1 to dodając kolejnego masz 2 użytkowników o tym samy id tj. id=2